### PR TITLE
WIP: Add rpm to the test cases

### PR DESCRIPTION
**Pull request by fhajredini**
Signed-off-by: Fisnik Hajredini <fhajredini@luxoft.com>

### CODE REVIEW

**Comment by fhajredini** [comment]
the Rpm test suite contains `RpmBasicTest` and `RpmInstallRemoveTest` classes.  However `rpm.RpmInstallRemoveTest.test_check_rpm_install_removal_log_file_size` test fails. Therefore this test is not included.

**Comment by sashko** [comment]
Why does the test fail then?

**Comment by fhajredini** [comment]
Failed 
rpm.RpmInstallRemoveTest.test_check_rpm_install_removal_log_file_size (from rpm.RpmInstallRemoveTest-20180720135533)

Error Message
invalid literal for int() with base 10: 'grep: /var/log/messages: No such file or directory\n0'

Stacktrace

Traceback (most recent call last):
File "/home/yoctouser/pelux_yocto/build/../sources/poky/meta/lib/oeqa/core/decorator/__init__.py", line 32, in wrapped_freturn func(*args, **kwargs)

File "/home/yoctouser/pelux_yocto/build/../sources/poky/meta/lib/oeqa/core/decorator/__init__.py", line 32, in wrapped_freturn func(*args, **kwargs)

File "/home/yoctouser/pelux_yocto/sources/poky/meta/lib/oeqa/runtime/cases/rpm.py", line 141, in test_check_rpm_install_removal_log_file_sizeself.assertGreaterEqual(int(output), 80,

ValueError: invalid literal for int() with base 10: 'grep: /var/log/messages: No such file or directory\n0'

File: Jenkinsfile [hunk 1]
```diff
@@ -8,36 +8,39 @@
 // that one has to allocate a node first, which would stay busy until the end of
 // the parallel pipeline
 
-void buildOnYoctoNode(String variant, String image, boolean checkSmokeTests = false) {
+void buildOnYoctoNode(String variant, String image, boolean checkSmokeTests = false, boolean checkBitbakeTests=false) {
```
Comment (thenor) [inline Jenkinsfile]: I think I had issues before when I tried to use more than one optional. If there is one out of two unset variables in a row it should be impossible for the script to know which one is not set.

File: ci-scripts/yocto.groovy [hunk 2]
```diff
@@ -216,8 +236,15 @@ void buildManifest(String variantName, String imageName, boolean smokeTests=fals
             boolean buildUpdate = variantName.startsWith("rpi")
             buildImageAndSDK(yoctoDir, imageName, buildUpdate)
             if (smokeTests) {
-                runSmokeTests(yoctoDir, imageName, testBitbake)
+                boolean weekly = env.WEEKLY_BUILD == "true"
+		runSmokeTests(yoctoDir, imageName)
+		if(weekly) {
+		    run runBitbakeTests(yoctoDir)
+		}				
+
```
Comment (thenor) [inline ci-scripts/yocto.groovy]: Yes, this is what I was thinking of as a solution.
The indentation and whitespace looks a bit strange though.

Comment (fhajredini) [reply ci-scripts/yocto.groovy]: ok. Will fix it once im back. I was working on vim so didnt notice it.

File: ci-scripts/yocto.groovy [hunk 3]
```diff
@@ -131,6 +131,26 @@ void runSmokeTests(String yoctoDir, String imageName) {
...
+	stage("Perform Bitbake Testing"){
+	    vagrant("/vagrant/cookbook/yocto/runqemu-bitbake-test.sh)
+	}
+    } catch (e){
+	echo "Bitbake tests failed"
+	println(e.getMessage())
+    } finally{
+	stage("Publish smoke test results") {
+	    reportsDir="/vagrant/${archiveDir}/test_reports/bitbake_tests/"
+            vagrant("mkdir -p ${reportsDir}")
```
Comment (thenor) [inline ci-scripts/yocto.groovy]: The indentation looks a bit strange here also.

**Pull request closed without merging**